Support reading Vault's address from Agent's config file

vishalnayak opened the pull request.

briankassouf commented:
This looks good, but i think we should also be adding the TLS config options here too.

jefferai commented on command/agent.go:
```diff
@@ -206,6 +207,29 @@ func (c *AgentCommand) Run(args []string) int {
+	// Check if the `-address` flag was set on the command
+	var vaultAddrFlagSet bool
+	f.Visit(func(f *flag.Flag) {
+		if f.Name == "address" {
```
Since this is used in a coupe places can you make this a const a la the diff I shared with you yesterday?

vishalnayak replied:
Fixed.

jefferai commented on command/agent.go:
```diff
@@ -206,6 +207,29 @@ func (c *AgentCommand) Run(args []string) int {
+
+	// If Vault address is not supplied via the command flag, refer to env var
+	// and config field in decresing order of preference.
+	if !vaultAddrFlagSet {
```
Probably better as a `switch` statement.

vishalnayak replied:
Fixed.

calvn commented on command/agent/config/config.go:
```diff
@@ -107,9 +112,35 @@ func LoadConfig(path string, logger log.Logger) (*Config, error) {
 		return nil, errwrap.Wrapf("error parsing 'cache':{{err}}", err)
 	}
 
+	err = parseCache(&result, list)
```
```suggestion
	err = parseVault(&result, list)
```

vishalnayak replied:
I am genuinely confused about this. It was working without this change, I don't know how.

vishalnayak commented:
@briankassouf @jefferai The TLS options are now added and they go through the logic of env and flag checking before the config takes effect. I have added `ca_cert`, `ca_path` and `tls_skip_verify` options, thinking that they should be sufficient. LMK if you think otherwise.

briankassouf commented:
@vishalnayak client cert and client key is an important one to get in. A lot of users use client authentication with TLS
